### PR TITLE
add hw03

### PR DESCRIPTION
**Pull request by TiBulat**
Added task1-4 for hw03

### CODE REVIEW

File: HW03/tim_bulat/tim_bulat_hw_03_t_03.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+require 'time'
+
+class Homework3
+  def get_lines(log, key_word)
+    log.split("\n").select { |line| line != '' && line.include?(key_word) }
+  end
+
+  def calc_time(core_lines)
+    i = 0
+    res = []
```
Comment (alexshgov) [inline HW03/tim_bulat/tim_bulat_hw_03_t_03.rb:12]: Avoid using the temp variables `i` and `res`

File: HW03/tim_bulat/tim_bulat_hw_03_t_01.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class Homework3
+  def get_lines(log, key_word)
+    log.split("\n").select { |line| line != '' && line.include?(key_word) }
```
Comment (B0RlS) [inline HW03/tim_bulat/tim_bulat_hw_03_t_01.rb:5]: better to use `find` instead of `select`

Comment (B0RlS) [inline HW03/tim_bulat/tim_bulat_hw_03_t_01.rb:5]:
```suggestion
  def get_lines(log)
    log.split("\n").select { |line| line != '' && line.include?('error') }
```

File: HW03/tim_bulat/tim_bulat_hw_03_t_02.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class Homework3
+  IP_REGEX = /\b(?:\d{1,3}\.){3}\d{1,3}\b/
+  DATE_REGEX = /\[.*?\]/
+  TO_REGEX = /".*?"/
```
Comment (B0RlS) [inline HW03/tim_bulat/tim_bulat_hw_03_t_02.rb:6]:
```suggestion
  DATETIME = /\b(?:\d{1,3}\.){3}\d{1,3}\b/.freeze
  IP = /\[.*?\]/.freeze
  ADDRESS = /".*?"/.freeze
```

File: HW03/tim_bulat/tim_bulat_hw_03_t_02.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  def get_lines(log, key_word)
+    log.split("\n").select { |line| line != '' && line.include?(key_word) }
+  end
+
+  def task2(log)
+    err_log = get_lines(log, 'POST')
+
+    err_log.map do |line|
+      "DATE: #{line[DATE_REGEX].delete('[]')} "\
```
Comment (B0RlS) [inline HW03/tim_bulat/tim_bulat_hw_03_t_02.rb:16]: it's better to update a normal value than to remove brackets and stuff

File: HW03/tim_bulat/tim_bulat_hw_03_t_03.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    while i < core_lines.length - 1
+      res << (core_lines[i + 1] - core_lines[i]).to_s
+      i += 1
+    end
+    res
+  end
+
+  def task3(log)
+    core_lines = get_lines(log, 'core')
+
```
Comment (B0RlS) [inline HW03/tim_bulat/tim_bulat_hw_03_t_03.rb:22]:
```suggestion
```

File: HW03/tim_bulat/tim_bulat_hw_03_t_04.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Homework3
+  def get_lines(log, key_word)
+    log.split("\n").select { |line| line != '' && line.include?(key_word) }
+  end
```
Comment (B0RlS) [inline HW03/tim_bulat/tim_bulat_hw_03_t_04.rb:6]:
```suggestion
```
 u don't need it

**Pull request merged**